pkg/subscriptions/subscriptions-view.jsx

```diff
@@ -0,0 +1,247 @@
+
+        var icon = this.props.icon;
+        if (icon == 'waiting')
+            icon = <div className="spinner spinner-lg"></div>;
+        else if (icon == 'error')
+            icon = <div className="pficon pficon-error-circle-o"></div>;
+
+        return (
+            <div className={ curtains + " blank-slate-pf" }>
+                <div className="blank-slate-pf-icon">
+                    {icon}
+                </div>
+                {description}
+                {message}
+            </div>
+        );
+    }
+});
+
+/* Component to show a dismissable error, message as child text
+ * dismissError callback function triggered when the close button is pressed
+ */
+var DismissableError = React.createClass({
+    handleDismissError: function(e) {
+        // only consider primary mouse button
+        if (!e || e.button !== 0)
+            return;
+        if (this.props.dismissError)
+            this.props.dismissError();
+        e.stopPropagation();
+    },
+    render: function() {
+        return (
+            <div className="alert alert-danger alert-dismissable alert-ct-top">
+                <span className="pficon pficon-error-circle-o" />
+                <span>{this.props.children}</span>
+                <button type="button" className="close" aria-hidden="true" onClick={this.handleDismissError}>
+                    <span className="pficon pficon-close"/>
+                </button>
+            </div>
+        );
+    }
+});
+
+/* Show subscriptions status of the system, offer to register/unregister the system
+ * Expected properties:
+ * status        subscription status
+ * error        error message to show (in EmptyState if not connected, as a dismissable alert otherwise
+ * dismissError callback, triggered for the dismissable error in connected state
+ * register     callback, triggered when user clicks on register
+ * unregister   callback, triggered when user clicks on unregister
+ */
+var SubscriptionStatus = React.createClass({
+    handleRegisterSystem: function(e) {
+        // only consider primary mouse button
+        if (!e || e.button !== 0)
+            return;
+        this.props.register();
+        e.stopPropagation();
+    },
+    handleUnregisterSystem: function(e) {
+        // only consider primary mouse button
+        if (!e || e.button !== 0)
+            return;
+        this.props.unregister();
+        e.stopPropagation();
+    },
+    render: function() {
+        var errorMessage;
+        if (this.props.error) {
+            errorMessage = (
+                <DismissableError dismissError={this.props.dismissError}>{this.props.error}</DismissableError>
+            );
+        }
+
+        var label;
+        var action;
+        var note;
+        var isUnregistering = (this.props.status == "unregistering");
+        if (this.props.status == 'Unknown') {
+            label = <label>{ _("Status: System isn't registered") }</label>;
+            action = (<button title="Register"
```

Comment:
> Should this button be disabled while unregistering or already registering the system?

Registration happens in the dialog, so the button isn't visible.

When unregistering, it isn't present - only `Unregister` is.